sale_weekday_delivery/models/sale_order.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+
+from .res_partner import WEEKDAY_MAPPING
+
+
+class SaleOrder(models.Model):
+
+    _inherit = "sale.order"
+
+    @api.depends(
+        "order_line.customer_lead",
+        "date_order",
+        "order_line.state",
+        "picking_policy",
+        "partner_shipping_id.delivery_schedule",
+        "partner_shipping_id.delivery_schedule_monday",
+        "partner_shipping_id.delivery_schedule_tuesday",
+        "partner_shipping_id.delivery_schedule_wednesday",
+        "partner_shipping_id.delivery_schedule_thursday",
+        "partner_shipping_id.delivery_schedule_friday",
+        "partner_shipping_id.delivery_schedule_saturday",
+        "partner_shipping_id.delivery_schedule_sunday",
+    )
+    def _compute_expected_date(self):
+        """Add dependencies to consider fixed weekdays delivery schedule"""
+        return super()._compute_expected_date()
+
+    @api.onchange("commitment_date")
+    def _onchange_commitment_date(self):
+        """Warns if commitment date is not a preferred weekday for delivery"""
+        res = super()._onchange_commitment_date()
+        if res:
+            return res
+        if (
+            self.commitment_date
+            and self.partner_shipping_id.delivery_schedule == "fix_weekdays"
+        ):
+            raw_commitment_date_weekday = self.commitment_date.weekday()
+            commitment_date_weekday = WEEKDAY_MAPPING.get(
```

Comment:
see the change asked on the mapping -> you'll have translated string here

Reply:
I added a parameter to get the translated fields description for error messages :wink: